src/pygama/raw/data_decoder.py

```diff
@@ -46,50 +48,49 @@ def __init__(self, garbage_length=256, packet_size_guess=1024):
         self.garbage_table.add_field('garbage_code',
                                      lgdo.Array(shape=garbage_length, dtype='uint32'))
 
-
-    def get_key_list(self):
-        """ overload with list of keys for this decoder.
-        e.g. `return range(n_channels)`
-        The default version works for decoders with single / no keys
-        """
+    def get_key_list(self) -> list:
+        """Overload with list of keys for this decoder, e.g. ``return
+        range(n_channels)``.  The default version works for decoders with
+        single / no keys"""
         return [None]
 
+    def get_decoded_values(self, key: Any = None):
+        """Get decoded values (optionally for a given key, typically a channel).
 
-    def get_decoded_values(self, key=None):
-        """ Get decoded values (optionally for a given key, typically a channel)
-
-        Must overload for your decoder if it has key-specific decoded
-        values. Note: implement key = None returns a "default" decoded_values
-
-        Otherwise, just returns self.decoded_values, which should be defined in
-        the constructor
+        .. note::
+            Must overload for your decoder if it has key-specific decoded
+            values. Must also implement ``key = None`` returns a "default"
+            ``decoded_values``. Otherwise, just returns
+            :attr:`self.decoded_values`, which should be defined in the
+            constructor.
         """
         if key is None:
             return self.decoded_values if hasattr(self, 'decoded_values') else None
 
         raise NotImplementedError("you need to implement key-specific get_decoded_values for", type(self).__name__)
 
+    def make_lgdo(self, key: Any = None, size: int = None):
```

Comment:
same question here on setting the int `size` to `None`

Reply:
Since `None` is already specified as default value, writing `size: int | None = None` seems a bit redundant to me. My philosophy is that, since type annotations are not checked by the interpreter, we should try to privilege readability for the user...

We might revise all this once (and if...) we'll have static type checkers like MyPy running as part of our CI.

Reply:
Since we're on this function declaration: should we be more restrictive than `Any` for the key type? What would you put there?

Reply:
Sounds good on not working about int vs None.

As long as its just a suggestion to the user, I would propose `int | str`. In principle a user could use anything, but we want to be able to key dictionaries and name HDF5 datasets using `key`, so as a suggestion to the user and int or a str is heavily preferred over other options.